src/nginx/t/init_service_configs_multiple.t

```diff
@@ -0,0 +1,294 @@
+$t->run();
+
+################################################################################
+
+my $response = ApiManager::http_get(
+  $NginxPort, '/shelves?key=this-is-an-api-key' );
+
+is($t->waitforfile("$t->{_testdir}/${report_done}"), 1, 'Report body file ready.');
+
+$t->stop_daemons();
+
+my ( $response_headers, $response_body ) = split /\r\n\r\n/, $response, 2;
+
+like( $response_headers, qr/HTTP\/1\.1 200 OK/, 'Returned HTTP 200.' );
+is( $response_body, <<'EOF', 'Shelves returned in the response body.' );
+{ "shelves": [
+    { "name": "shelves/1", "theme": "Fiction" },
+    { "name": "shelves/2", "theme": "Fantasy" }
+  ]
+}
+EOF
+
+my @requests = ApiManager::read_http_stream( $t, 'bookstore.log' );
+is( scalar @requests, 1, 'Backend received one request' );
+
+my $r = shift @requests;
+
+is( $r->{verb}, 'GET', 'Backend request was a get' );
+is( $r->{uri}, '/shelves?key=this-is-an-api-key', 'Backend uri was /shelves' );
+is( $r->{headers}->{host}, "127.0.0.1:${BackendPort}", 'Host header was set' );
+
+@requests = ApiManager::read_http_stream( $t, 'servicecontrol.log' );
+is( scalar @requests, 3, 'Service control received three requests' );
+
+# check
+$r = shift @requests;
+is( $r->{verb}, 'POST', ':check verb was post' );
+is( $r->{uri}, '/v1/services/endpoints-test.cloudendpointsapis.com:check',
+  ':check was called');
+is( $r->{headers}->{host}, "127.0.0.1:${ServiceControlPort}",
+  'Host header was set');
+is( $r->{headers}->{'content-type'}, 'application/x-protobuf',
+  ':check Content-Type was protocol buffer');
+
+# test allocateQuota request was requested
+$r = shift @requests;
+is( $r->{verb}, 'POST', ':allocateQuota verb was post' );
+is( $r->{uri},
+  '/v1/services/endpoints-test.cloudendpointsapis.com:allocateQuota',
+  ':allocateQuota was called');
+is( $r->{headers}->{host}, "127.0.0.1:${ServiceControlPort}",
+  'Host header was set');
+is( $r->{headers}->{'content-type'}, 'application/x-protobuf',
+  ':check Content-Type was protocol buffer' );
+
+my $allocate_quota_request = decode_json(ServiceControl::convert_proto(
+  $r->{body}, 'quota_request', 'json' ) );
+
+my @quotaMetrics =
+  @{ $allocate_quota_request->{allocateOperation}->{quotaMetrics} };
+is( @quotaMetrics, 2, "Quota metrics should have two elements" );
+
+my @sorted_quotaMetrics =
+  sort { $a->{metricName} cmp $b->{metricName} } @quotaMetrics;
+
+is( $sorted_quotaMetrics[0]->{metricName}, "metrics_first",
+  "Quota metric name is 'metrics_first'" );
+is( $sorted_quotaMetrics[0]->{metricValues}[0]->{int64Value}, 2,
+  "Quota metric value is 2" );
+is( $sorted_quotaMetrics[1]->{metricName}, "metrics_second",
+  "Quota metric name is 'metrics_second'" );
+is( $sorted_quotaMetrics[1]->{metricValues}[0]->{int64Value}, 1,
+  "Quota metric value is 1" );
+
+# check report
+$r = shift @requests;
+
+is( $r->{verb}, 'POST', ':report verb was post' );
+is( $r->{uri}, '/v1/services/endpoints-test.cloudendpointsapis.com:report',
+  ':report was called');
```

Comment:
can we verify config_id in the Check, Quota and Report call?
Since the first one is 80%,   for sure, the first one will be picked.